src/types/label.js

```diff
@@ -149,7 +149,7 @@ function calculatePosition(start, size, adjust, position) {
   } else if (position === 'end') {
     return start - size + adjust;
   }
-  return start - size / 2 + adjust;
+  return start + getSize(size, position, size / 2) + adjust ;
```

Comment:
I made a sign error?
```suggestion
  return start - getSize(size, position, size / 2) + adjust ;
```